File: count-items-matching-a-rule-2.py
# define class Solution
class Solution:
    # define function countMatches, which takes in the class instance (self), a list of lists of
    # strings called items, a string called ruleKey, and a string called ruleValue, and returns an
    # integer.
    def countMatches(self, items: List[List[str]], ruleKey: str, ruleValue: str) -> int:
        # create a dictionary called rule_maps with some initialized keys and values.
        rule_maps = {
            "type": 0,
            "color": 1,
            "name": 2
        }
        # create variable count and initialize as 0.
        count = 0
        # loop through each item in items.
        for item in items:
            # the value of item at key rule_maps at key ruleKey is equal to ruleValue
            if item[rule_maps[ruleKey]] == ruleValue:
                # increment count by 1
                count += 1
        # return count
        return count

File: count-items-matching-a-rule-2.ts
// declare function countMatches, which takes in an array of arrays of strings called items, a
// string called ruleKey, and a string called ruleValue, and returns a number.
function countMatches(
  items: string[][],
  ruleKey: string,
  ruleValue: string
): number {
  // create mutable variable count and initialize it to 0
  let count = 0;

  // loop through each item and pull off the three indices of the item.
  // these indices are locally called type, color, and name.
  for (const [type, color, name] of items) {
    // create a switch based on the value of ruleKey
    switch (ruleKey) {
      // if the ruleKey is type
      case "type":
        // if type is equal to ruleValue
        if (type == ruleValue) {
          // increment count
          count++;
        }
        break;
      // if the ruleKey is color
      case "color":
        // if color is equal to ruleValue
        if (color == ruleValue) {
          // increment count
          count++;
        }
        break;
      // if the ruleKey is name
      case "name":
        // if name is equal to ruleValue
        if (name == ruleValue) {
          // increment count
          count++;
        }
        break;
    }
  }

  return count;
}
